docs/features/non-source-outputs.md

```diff
@@ -0,0 +1,141 @@
+
+In order to decide if a file is modified the new target will require a task that can decide if a file has been modified since the previous build. This logic is fairly simple, as it just has to compare the timestamp of the output list file with the items listed within, but it's worth calling out that this does duplicate some of the `MSBuild` logic in the compiler tasks. The concept of dynamic inputs/outputs is proposed by `MSBuild` but not currently implemented. If this feature is added we could remove this duplication.
+
+## Open Questions
+
+- Naming: This proposal originally used the term `NonSource` but that's not particularly descriptive and perhaps odd in that it's not actually the mirror of a source output in that it's not going into the compilation. `Artifact` seems like a better term? Or simply `File`?
+- Do we want custom mapping for the `compiler_artifact.` scheme, or could we just use the existing `build_metadata.<ItemType>.<Metadata>` mechanism and recognize `CompilerMappedArtifact` as a special key?
+  - This is doable, but does tie the MSBuild item type name directly into the compiler, rather than having a level of indirection.
+- File mapping: should we just let the generators party on the file system and allow absolute paths and `.`, `..` etc, rather than having the host control the output location?
+  - An issue with this approach is that a user can control their project layout via MSBuild, and the generator won't know that it has changed, meaning you might end up with files in odd places
+  - It would be possible to route though e.g. `OutDir` either automatically, or by the generator author specifying it as a `CompilerVisibleProperty` but this relies on the author doing the correct thing (and multiple authors having to copy the same logic)
+  - It seems better to write to a 'virtual' location, and let the host re-direct that as appropriate.
+- File mapping: within a hint name should we allow sub-directories? So that a given generator can write to e.g. `<base>/css` and `<base>/js` and just map the `<base>` directory to wherever is appropriate?
+- File mapping: can we just have the compiler write them to the obj directory (or the `GeneratedFilesOut` directory) and use MSBuild to do the copying? Either by default or require the author to supply targets to do the copying after build?
+  - This makes things for the compiler simpler, but doesn't optimize for disk hits or consistency. A given generator may not even _know_ what its outputs are going to be ahead of time, as the artifact name may be derived from a user input, making it almost impossible for the generator to provide the copy
+    - We could say that each generators outputs go to a specific folder, like the generated files do, so that it could just copy `<dir>/*` to the location that matters.
```

Comment:
This still requires authors to ship extra MSBuild logic for the copying which seems undesirable.